esmvaltool/diag_scripts/lifetime/lifetime_base.py

```diff
@@ -0,0 +1,875 @@
+            ],
+            'references': [
+                'acknow_project',
+            ],
+            'ancestors': ancestor_files,
+            **kwargs
+        }
+        return record
+
+    def get_plot_path(self, plot_type, var_info, add_ext=True):
+        """Get plot full path from variable info.
+
+        Parameters
+        ----------
+        plot_type: str
+            Name of the plot
+        var_info: dict
+            Variable information from ESMValTool
+        add_ext: bool, optional (default: True)
+            Add filename extension from configuration file.
+
+        """
+        return os.path.join(
+            self.get_plot_folder(var_info),
+            self.get_plot_name(plot_type, var_info, add_ext=add_ext),
+        )
+
+    def get_plot_folder(self, var_info):
+        """Get plot storage folder from variable info.
+
+        Parameters
+        ----------
+        var_info: dict
+            Variable information from ESMValTool
+
+        """
+        info = {
+            'real_name': self._real_name(var_info['variable_group']),
+            **var_info
+        }
+        folder = os.path.expandvars(
+            os.path.expanduser(
+                list(_replace_tags(self.plot_folder, info))[0]
+            )
+        )
+        if self.plot_folder.startswith('/'):
+            folder = '/' + folder
+        if not os.path.isdir(folder):
+            os.makedirs(folder, exist_ok=True)
+        return folder
+
+    def get_plot_name(self, plot_type, var_info, add_ext=True):
+        """Get plot filename from variable info.
+
+        Parameters
+        ----------
+        plot_type: str
+            Name of the plot
+        var_info: dict
+            Variable information from ESMValTool
+        add_ext: bool, optional (default: True)
+            Add filename extension from configuration file.
+
+        """
+        info = {
+            "plot_type": plot_type,
+            'real_name': self._real_name(var_info['variable_group']),
+            **var_info
+        }
+        file_name = list(_replace_tags(self.plot_filename, info))[0]
+        if add_ext:
+            file_name = self._add_file_extension(file_name)
+        return file_name
+
+    @staticmethod
+    def _set_rasterized(axes=None):
+        """Rasterize all artists and collection of axes if desired."""
+        if axes is None:
+            axes = plt.gca()
+        if not isinstance(axes, list):
+            axes = [axes]
+        for single_axes in axes:
+            for artist in single_axes.artists:
+                artist.set_rasterized(True)
+            for collection in single_axes.collections:
+                collection.set_rasterized(True)
+
+    @staticmethod
+    def _real_name(variable_group):
+        for subfix in ('Ymean', 'Ysum', 'mean', 'sum'):
+            if variable_group.endswith(subfix):
+                variable_group = variable_group.replace(subfix, '')
+        return variable_group
```

Comment:
Please check if you really needs all these class methods, it seems to me like at least some of them are not used.